### DISCUSSION TITLE
Support for specifying an account id when querying for CloudWatch metrics from Grafana

### DISCUSSION THREAD

**Discussion by kevingerard72** [grafana/grafana#41818] [Data Sources]
AWS allows for sharing of metrics from one account to another.
When querying for metrics in CloudWatch, the UI allows for selecting a source account id.
![image](https://user-images.githubusercontent.com/81374351/113993927-3a23d100-9855-11eb-9c75-7a8ab4e7cd00.png)

Unfortunately, this functionality appears not to be available in Grafana.

When specifying a specific account ID, here is what the request from CloudWatch looks like:
_{
    "view": "timeSeries",
    "stacked": false,
    "metrics": [
        [ { "expression": "SEARCH('{AWS/RDS} MetricName=\"ActiveTransactions\"', 'Average', 300)", "id": "e1", "period": 300, "accountId": "############" } ]
    ],
    "region": "eu-west-1"
}_

Notice the "accountId" entry in the metrics portion of the request.

I have found no way of including the account ID when querying from Grafana.
Am I missing something?
If indeed this is not available, can it be added?

Many thanks

**Comment by sunker** [comment]
Thanks for reporting this @kevingerard72! Didn't know that features existed in the CloudWatch console, but I think it sounds great. The questions is how to solve this though. I might be possible to do using data links. I'll do some investigating on my side and report back here.

**Comment by kevingerard72** [comment]
Hi Eric,
Thanks for your feedback.
Feel free to reach out to me if you need more info or screenshots.
Cheers,
Kevin

**Comment by gigawatts** [comment]
I was just about to open a new issue asking if exactly this was possible. I'll be eagerly awaiting this feature!

**Comment by kevingerard72** [comment]
Hi Erik,
Do you have an update for us related to this issue?
Many thanks,
Kevin

**Comment by sunker** [comment]
Hi @kevingerard72! I'm talking to AWS engineers about this. I'm afraid it might take some more time to investigate this.

**Comment by mourya92** [comment]
+1 to the ask .. my usecase requires me to provide an `accountId` input to the query.

**Comment by mrh666** [comment]
+1 to have an accountId really will be great!

**Comment by duckaraptor** [comment]
+1 because of the requirement I have. @sunker do you have any update on this?
Thanks again for working on this feature.

**Comment by sarahzinger** [comment]
Hey all, it's been a minute but I wanted to make sure you all got a chance to see and test out our support for [Cross Account Queries with Cloudwatch](https://github.com/grafana/grafana/blob/a7cec6dd03273f1546f9c3a9b70fdc52960a320a/docs/sources/datasources/aws-cloudwatch/query-editor/index.md?plain=1#L217), I think it should address some of your concerns:
<img width="1203" alt="Screenshot 2023-05-18 at 4 41 21 PM" src="https://github.com/grafana/grafana/assets/6620164/2b2585d1-c5f8-4644-93f4-cbeb654c662c">
